ctapipe/image/extractor.py

```diff
@@ -345,6 +345,94 @@ def integration_correction(
     return correction
 
 
+def time_parameters(waveforms, upper_limit, lower_limit, peak_index=-1):
+    """
+    Calculates the full width at half maximum (fwhm), rise time, and fall time of waveforms.
+    
+    Parameters
+    ----------
+    waveforms : ndarray
+        Waveforms stored in a numpy array.
+        Shape: (n_pix, n_samples)
+    peak_index : ndarray
+        Peak index of waveforms
+        Shape: (n_pix, )
+    upper_limit : float
+        Upper fraction of peak maximum
+    lower_limit : float
+        Lower fraction of peak maximum
+
+    Returns
+    -------
+    fwhm : list of int
+        Full width half maximum of a pulse.
+        Shape : (n_pix)
+    rise_time : list of int
+        Rise time of the pulse
+        Shape : (n_pix)
+    fall_time : list of int
+        Fall time of the pulse
+        Shape : (n_pix)
+
+    """
+
+    if peak_index == -1:  # take the maximum as a default
+        peak_index = np.argmax(waveforms, axis=-1)
+
+    nsamples = len(waveforms)
+
+    fwhm = []
+    rise_time = []
+    fall_time = []
+
+    for i in prange(0, nsamples):
```

Comment:
prange implies `numba`? But the function is not decorated with `njit`

Reply:
We also found in the past, that doing things in parallel at the waveform level is not actually faster than just using 1 core, probably due to the cache / data moving overhead.